### ISSUE TITLE
pagination problem

### ISSUE THREAD

**Issue by martianoff** [alfajango/jquery-dynatable#113]
Code:

$(function() {
                    $('#my-table1').dynatable({
                        dataset: {
                            ajax: true,
                            ajaxUrl: '/lookup2.php?req=json&source=var1&keyword=var2',
                            ajaxOnLoad: true,
                            perPageDefault: 100,
                            perPageOptions: [50, 100, 200],
                            records: []
                        },
                        features: {
                            PushState: true,
                            paginate: true,
                            search: false,
                            sort: false
                        }
                    }); 
                }); 

Server return:
{"records": (......), "totalRecordCount": 124, "queryRecordCount": 100}

What I expect: 2 pages with 100 + 24 results
What I have: only 1 page, bottom text: Showing 100 of 100 records (filtered from 124 total records)

**Comment by pandorawombat** [comment]
Did anyone ever look into this? I am having the exact same issue and can't find a solution that works. I am correctly passing the number of records queries and the total, but it is showing (x records filtered from y records) but not giving any pagination to allow queries for those additional records. It is instead putting up a one page pagination square with forward and back arrows disabled.

**Comment by pandorawombat** [comment]
Actually, msksimru's fix (it was line 1334 in my source) fixed it, but I wonder if that fix will break anything else?

**Comment by pandorawombat** [comment]
Yes, it does break other stuff to change that line. And it still didn't fix the whole thing. So I put it back. Can anyone tell me how to properly fix this?

Pagination looks like this:
![badpagination](https://cloud.githubusercontent.com/assets/7469486/8770226/33b5642c-2e7b-11e5-90db-1fc2e9adcd63.png)
with forward and back arrows disabled, instead of paginating and saying 1 to 5 of 2938 which is what I would expect.

example JSON is as follows (the JSON will return subsets of 5 or 20 records based on the query sent by the pagination, this is the "opening" query of exactly 5)
{"records":[{"id":"10335","name":"Jacqueline Raque","email":"raquej11@gmail.com","phone":"","address":"","color":"d9534f","image":"\/app\/assets\/images\/user_placeholder.png"},{"id":"10336","name":"Dan Cassin","email":"","phone":"502 554-2397","address":"","color":"d9534f","image":"\/app\/assets\/images\/user_placeholder.png"},{"id":"10337","name":"Recipe Recipe","email":"","phone":"","address":"","color":"d9534f","image":"\/app\/assets\/images\/user_placeholder.png"},{"id":"10338","name":"David Fonseca","email":"dfonseca@semonin.com","phone":"","address":"","color":"d9534f","image":"\/app\/assets\/images\/user_placeholder.png"},{"id":"10339","name":"Ruth Flath","email":"rflath@semonin.com","phone":"","address":"","color":"d9534f","image":"\/app\/assets\/images\/user_placeholder.png"}],
"queryRecordCount":"5",
"totalRecordCount":2938}

Here's the init code, which will look complex because we are using a writer to create a special format, but I tried the exact same ajax JSON data in a plain old table with no special writer, and it still had the pagination wrong:

$('ul#contact-list.list-group').dynatable({
                    dataset: {
                        ajax: true,
                        ajaxUrl: '/ajax/contact_json/',
                        ajaxOnLoad: true,
                        perPageDefault: 5,
                        perPageOptions: [5,20],
                        records: []
                    },
                    table: {
                        bodyRowSelector: '.a-contact'
                    },
                    features: {
                        paginate: true,
                        sort: true,
                        search: true,
                        pushState: false,
                        recordCount: true,
                        perPageSelect: true
                    },
                    inputs: {
                        queryEvent: 'keyup',
                        queries: jQuery('#contact-search'),
                        processingText: '{{ $loaderHtml }}', 
                        paginationClass: 'pagination pagination-sm',
                        paginationActiveClass: 'active',
                        paginationDisabledClass: 'disabled',
                        perPageTarget: '#contact-perpage',
                        perPagePlacement: 'append',
                        recordCountTarget: '#contact-pagination',
                        recordCountPlacement: 'append',
                        paginationLinkTarget: '#contact-pagination',
                        paginationLinkPlacement: 'prepend',
                        paginationPrev: '&laquo;',
                        paginationNext: '&raquo;',
                    },
                    readers: {
                        _rowReader: ulReader
                    },
                    writers: {
                        _rowWriter: ulWriter
                    },
                    params: {
                        
                    }
                });

Do I need to add some params to this init? Is there something missing from my JSON data? I have been beating my head against this for quite a while and have tried things I read in several threads, to no avail.
